File: src/jbased/ActivationRelu.java
package jbased;

public class ActivationRelu {
    double[][] output;

    public void forward(double[][] inputs) {
        for (int i = 0; i < inputs.length; i++) {
            for (int j = 0; j < inputs[0].length; j++) {
                if (inputs[i][j] < 0) inputs[i][j] = 0;
            }
        }
        output = inputs;
    }
}

File: src/kbased/ActivationRelu.kt
package kbased

class ActivationRelu {
    lateinit var output: Array<DoubleArray>
    fun forward(inputs: Array<DoubleArray>) {
        for (i in inputs.indices) {
            for (j in inputs[0].indices) {
                if (inputs[i][j] < 0) inputs[i][j] = 0.0
            }
        }
        output = inputs
    }
}